src/core_modules/capture-core/components/Pages/NewRelationship/RegisterTei/RegistrationSection/ProgramSelector/ProgramSelector.container.js

```diff
@@ -6,7 +6,7 @@ import { changeProgram, clearProgramFilter } from '../registrationSection.action
 const mapStateToProps = (state: ReduxState) => ({
     orgUnitIds: state.newRelationshipRegisterTei.orgUnit ? [state.newRelationshipRegisterTei.orgUnit.id] : null,
     trackedEntityTypeId: state.newRelationship.selectedRelationshipType.to.trackedEntityTypeId,
-    value: state.newRelationshipRegisterTei.programId,
+    value: state.newRelationshipRegisterTei.programId ?? state.newRelationship?.selectedRelationshipType?.to?.programId,
```

Comment:
We should try to figure out why state.newRelationshipRegisterTei.programId is not updated instead, due to a couple of reasons: a) The programId we are looking for could be on the "from" side (this workflow is not working properly at the moment, created JIRA issue for this) b) There is some logic that relies on `state.newRelationshipRegisterTei.programId` being set.

Reply:
thanks @JoakimSM I agree. on the second look I found that the programId is set to null here https://github.com/dhis2/capture-app/pull/2435/files#diff-182458f2805a717c67f38531199e47402242a16c155da4562cd53071ec1b15beL75. 
So putting the programId there should fix it